include/aws/s3/private/s3_meta_request_impl.h

```diff
@@ -25,115 +26,35 @@ struct aws_http_make_request_options;
-    struct aws_ref_count ref_count;
-
-    struct aws_allocator *allocator;
-
-    /* Owning meta request. */
-    struct aws_s3_meta_request *meta_request;
-
-    /* Request body to use when sending the request. The contents of this body will be re-used if a request is
-     * retried.*/
-    struct aws_byte_buf request_body;
-
-    /* Part number that this request refers to.  If this is not a part, this can be 0.  (S3 Part Numbers start at 1.)
-     * However, must currently be a valid part number (ie: greater than 0) if the response body is to be streamed to the
-     * caller.
-     */
-    uint32_t part_number;
-
-    /* Tag that defines what the built request will actually consist of.  This is meant to be space for an enum defined
-     * by the derived type.  Request tags do not necessarily map 1:1 with actual S3 API requests.  For example, they can
-     * be more contextual, like "first part" instead of just "part".) */
-    /* TODO we could potentially combine these with the bitfields below. */
-    int request_tag;
-
-    /* When true, response headers from the request will be stored in the request's response_headers variable. */
-    uint32_t record_response_headers : 1;
+enum aws_s3_meta_request_update_flags {
+    /* There are no connections available. Meta request can wait for anything in progress to finish up, but shouldn't
+       expect that there will be any connections for issuing requests. */
+    AWS_S3_META_REQUEST_UPDATE_FLAG_NO_ENDPOINT_CONNECTIONS = 0x00000001,
 
-    /* When true, the response body will be streamed back to the caller. */
-    uint32_t stream_response_body : 1;
-
-    /* When true, the response body buffer will be allocated in the size of a part. */
-    uint32_t part_size_response_body : 1;
-
-    uint32_t request_was_sent : 1;
-
-    /* Members of this structure will be repopulated each time the request is sent.  For example, If the request fails,
-     * and needs to be retried, then the members of this structure will be cleaned up and re-populated on the next send.
-     */
-    struct {
-
-        /* The HTTP message to send for this request. */
-        struct aws_http_message *message;
-
-        /* Signable created for the above message. */
-        struct aws_signable *signable;
-
-        /* Recorded response headers for the request. Set only when the request desc has record_response_headers set to
-         * true or when this response indicates an error. */
-        struct aws_http_headers *response_headers;
-
-        /* Recorded response body of the request. */
-        struct aws_byte_buf response_body;
-
-        /* Returned response status of this request. */
-        int response_status;
-
-    } send_data;
+    /* The client potentially has multiple meta requests that it can spread across connections, and the given meta
+       request can selectively not return a request if there is a performance reason to do so.*/
+    AWS_S3_META_REQUEST_UPDATE_FLAG_CONSERVATIVE = 0x00000002,
```

Comment:
I feel like `_THROTTLED` might be more descriptive than `_CONSERVATIVE`, and then maybe have a version that is `_ALL` to differentiate?

Reply:
Yeah, I don't love using "conservative" either. I was originally thinking about something similar to "throttled" because "conservative" is really generic.  

But we don't necessarily have to do any type of request limiting outside of auto-ranged-gets, so it wasn't always applicable. :-/  I was looking for something that said, "Assume we have more meta requests that have work right now, so be careful when spinning up additional requests".